### PR TITLE
Added note about GRUB and PXE

### PR DESCRIPTION
**Pull request by kayrus**
split https://github.com/coreos/ignition/pull/159 into two parts.

### CODE REVIEW

File: ignition/what-is-ignition.md [hunk 1]
```diff
@@ -66,7 +60,9 @@ experince crippled because the metadata for their platform isn't supported. It
 is possible to write a [custom metadata agent][custom agent] to fetch the
 necessary data.
 
-[custom agent]: examples.md#custom-metadata-agent
+### How Ignition is Being Executed ##
```
Comment (crawford) [inline ignition/what-is-ignition.md]: ### When is Ignition Executed ###

File: ignition/what-is-ignition.md [hunk 2]
```diff
@@ -27,8 +25,6 @@ octal or decimal. Ignition's configuration is also versioned which allows it to
 be improved in the future without having to worry as much about maintaining
 backward compatibility.
 
-[cloudinit]: https://github.com/coreos/coreos-cloudinit
-
 ### Ignition Only Runs Once ##
```
Comment (crawford) [inline ignition/what-is-ignition.md]: Can you fix up the trailing `#` while you are in here? The guidelines specify that these are no longer included.

File: ignition/what-is-ignition.md [hunk 3]
```diff
@@ -66,7 +60,9 @@ experince crippled because the metadata for their platform isn't supported. It
 is possible to write a [custom metadata agent][custom agent] to fetch the
 necessary data.
 
-[custom agent]: examples.md#custom-metadata-agent
+### How Ignition is Being Executed ##
+
+GRUB checks the disks for the dummy disk UUID and sets `coreos.first_boot=1` parameter accordingly. This parameter is being processed by [systemd-generators] every boot inside [initramfs] and whenever this parementer is set systemd adds `ignition.target` as one of the `initrd.target` requirement to run Ignition. In case when you use [PXE][supported platforms] deployment `coreos.first_boot=1` parameter needs to be set manually otherwise Ignition won't run.
```
Comment (crawford) [inline ignition/what-is-ignition.md]: On boot, GRUB checks the disks for the dummy disk UUID (`00000000-0000-0000-0000-000000000000`) and sets `coreos.first_boot=1` if it is found. This parameter is then processed by a [systemd-generator] in the [initramfs] and if the parameter is set to non-zero, `ignition.target` is set as a dependency of `initrd.target`, causing Ignition to run.

Note that [PXE][supported platforms] deployments don't using GRUB to boot, so the `coreos.first_boot=1` parameter will need to be added to the boot arguments in order for Ignition to run.

**Comment by kayrus** [comment]
@crawford Done

**Comment by crawford** [comment]
LGTM

**Pull request merged**